project/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -40,13 +45,27 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.ResponseHandler;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
 
 /** 
  * Queries the backend database and serves location-based search results of (brief)
  * election/candidate information for the directory page.
  */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  private final static String VOTER_INFO_QUERY_URL =
+      String.format("https://www.googleapis.com/civicinfo/v2/voterinfo?key=%s", Config.CIVIC_INFO_API_KEY);
+  private final static String RELEVANT_NONSPECIFIC_ADDRESS_ALERT =
+      "Your input address was not specific enough. We are providing all possible" +
+      " elections that may be relevant.";
```

Comment:
It's also possible that the input address was specific, but a non-residential address.